### PR TITLE
notifications: add comment notification for subcoms

### PR DESCRIPTION
**Pull request by carlinmack**
:heart: Thank you for your contribution!

Close https://github.com/inveniosoftware/invenio-requests/issues/417

### Description

> Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_communities/notifications/builders.py [hunk 1]
```diff
@@ -231,11 +231,6 @@ class SubComInvitationBuilderBase(SubCommunityBuilderBase):
         EntityResolve("executing_user"),
     ]
 
-    recipients = [
-        CommunityMembersRecipient("request.created_by", roles=["owner", "manager"]),
-        CommunityMembersRecipient("request.receiver", roles=["owner", "manager"]),
-    ]
-
```
Comment (carlinmack) [inline invenio_communities/notifications/builders.py:238]: this is inherited from `SubCommunityBuilderBase`

File: invenio_communities/templates/semantic-ui/invenio_notifications/comment-subcommunity-invitation-request.create.jinja [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{% set invenio_request = notification.context.request %}
+{% set invenio_request_event = notification.context.request_event %}
+
+{# created_by is either a resolved user or an email (for guests) #}
+{% set event_creator_name = invenio_request_event.created_by.username or invenio_request_event.created_by %}
+{% set request_id = invenio_request.id %}
+{% set request_event_content = invenio_request_event.payload.content | safe %}
+{% set request_title = invenio_request.title | safe %}
+{% set child_community = invenio_request.receiver.slug %}
```
Comment (carlinmack) [inline invenio_communities/templates/semantic-ui/invenio_notifications/comment-subcommunity-invitation-request.create.jinja:9]: this is the only line different between the two functions

**Comment by ntarocco** [comment]
Is this fixing somehow the notifications for comments in an inclusion/review request for a community too (not only sub-community)?

**Comment by slint** [comment]
> Is this fixing somehow the notifications for comments in an inclusion/review request for a community too (not only sub-community)?

These are defined in `invenio-rdm-records`, but the fact that we now have the `comment_notification_builder` attribute on the request, means that we can override it per-request type and adjust recipients based on their type (users or communities).

**Pull request merged**